File: android/src/main/java/expo/modules/stickyview/ExpoStickyViewModule.kt
package expo.modules.stickyview

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoStickyViewModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoStickyView")

    View(ExpoStickyView::class) {
      Events("onStickyChange")
      Prop("topOffset") { view: ExpoStickyView, top: Double? ->
        if (top != null) {
          view.setStickyTop(top)
        } else {
          view.clearSticky()
        }
      }
      Prop("bottomOffset") { view: ExpoStickyView, bottom: Double? ->
        if (bottom != null) {
          view.setStickyBottom(bottom)
        } else {
          view.clearSticky()
        }
      }
    }
  }
}

File: ios/ExpoStickyViewModule.swift
import ExpoModulesCore

public class ExpoStickyViewModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoStickyView")

    View(ExpoStickyView.self) {
      Events("onStickyChange")

      Prop("topOffset") { (view: ExpoStickyView, top: Double?) in
        if let top = top {
          view.setStickyTop(top)
        }
      }

      Prop("bottomOffset") { (view: ExpoStickyView, bottom: Double?) in
        if let bottom = bottom {
          view.setStickyBottom(bottom)
        }
      }
    }
  }
}